### PR TITLE
 Configurations for EI 6.4.0 - pattern-2 deployment 

### PR DESCRIPTION
**Pull request by Savidude**
PCF Ops Manager v2.4-build.131
tile version 12.0.8
bosh version 5.3.1-8366c6fd-2018-09-25T18:25:51Z
pcf version 12.0.8

### CODE REVIEW

File: pattern-2/bosh-release/jobs/wso2ei/spec [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  type: peers
+- name: wso2ei_analytics_worker
+  type: worker-node
+
+provides:
+- name: wso2ei
+  type: wso2ei-node
+  properties:
+  - route_registrar.routes
+- name: peers
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei/spec]: Shall we change `peers` to a more specific name? eg: `integrator-peers`

File: pattern-2/bosh-release/jobs/wso2ei/templates/ctl.erb [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei/templates/ctl.erb]: Shall we update the license here?

File: pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/spec [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+name: wso2ei_analytics_dashboard
+templates:
+  ctl.erb: bin/ctl
+  conf/deployment.yaml: wso2/analytics/conf/dashboard/deployment.yaml
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/spec]: Shouldn't this be an embedded ruby file?

File: pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/templates/conf/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+#        jdbcUrl: 'jdbc:h2:${sys:carbon.home}/wso2/worker/database/IS_ANALYTICS_DB;AUTO_SERVER=TRUE'
+#        username: wso2carbon
+#        password: wso2carbon
+#        driverClassName: org.h2.Driver
+#        maxPoolSize: 50
+#        idleTimeout: 60000
+#        validationTimeout: 30000
+#        isAutoCommit: false
+
+
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/templates/conf/deployment.yaml]: Shall we remove this extra line?

File: pattern-2/bosh-release/jobs/wso2ei_analytics_worker/spec [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+name: wso2ei_analytics_worker
+templates:
+  ctl.erb: bin/ctl
+  conf/deployment.yaml: wso2/analytics/conf/worker/deployment.yaml
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_worker/spec]: deployment.yaml should be an embedded ruby file.

File: pattern-2/bosh-release/jobs/wso2ei_analytics_worker/spec [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    description: WSO2 EI Analytics database database driver name
+  wso2ei.sp_cluster_db.query:
+    description: WSO2 EI Analytics database database validation query
+  wso2ei.sp_cluster_db.username:
+    description: WSO2 EI Analytics database username
+  wso2ei.sp_cluster_db.password:
+    description: WSO2 EI Analytics database password
+
+
+  route_registrar.routes:
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_worker/spec]: This property is not used anywhere. Shall we remove this?

File: pattern-2/tile/tile.yml [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+    max_in_flight: 1
+    properties:
+      wso2ei:
+        sp_cluster_db:
+          jdbc_url: (( .properties.sp_db_jdbc_url.value ))
+          driver: (( .properties.sp_db_driver.value ))
+          query: (( .properties.sp_db_query.value ))
+          username: (( .properties.sp_db_username.value ))
+          password: (( .properties.sp_db_password.value ))
+      route_registrar:
```
Comment (VimukthiPerera) [inline pattern-2/tile/tile.yml]: As this property is not used, shall we remove it?

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: pattern-2/bosh-release/jobs/health_check/templates/run.sh.erb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    fi
+}
+
+healthCheckEP=<%= p("health_check.endpoint") %>
+COUNTER=0
+
+# While the endpoint is not alive, and the server has been retrying for less than 3 minutes
+while [ ${isAlive} -eq 0 ]&&[ ${COUNTER} -lt 18 ]; do
+    sleep 10s
+    health_check ${healthCheckEP} "<%= p("health_check.name") %>"
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/health_check/templates/run.sh.erb:33]: Shall we add the same change for pattern-1 as well?

Comment (Savidude) [reply pattern-2/bosh-release/jobs/health_check/templates/run.sh.erb:33]: Added to Pattern 1 https://github.com/wso2/pivotal-cf-ei/pull/7/commits/adf8bf3100d136f254abbf040476814fce49f8a3

**Review by VimukthiPerera — APPROVED** [review]

File: pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/templates/conf/deployment.yaml.erb [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+#        username: wso2carbon
+#        password: wso2carbon
+#        driverClassName: org.h2.Driver
+#        maxPoolSize: 50
+#        idleTimeout: 60000
+#        connectionTestQuery: SELECT 1
+#        validationTimeout: 30000
+#        isAutoCommit: false
+
+
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_dashboard/templates/conf/deployment.yaml.erb]: An extra line here. Shall we remove this?

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: pattern-1/bosh-release/jobs/health_check/templates/run.sh.erb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+            echo "WSO2 EI $2 is Running!"
+            isAlive=1
+        fi
+    else
+        >&2 echo "WSO2 EI $2 is not alive. Retrying in 10s..."
+        isAlive=0
+    fi
+}
+
+carbonHealthCheckEP=<%= p("health_check.endpoint") %>
```
Comment (VimukthiPerera) [inline pattern-1/bosh-release/jobs/health_check/templates/run.sh.erb]: Shall we change this to a more generic name? Like `healthCheckEP`

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: pattern-2/bosh-release/jobs/wso2ei_analytics_worker/templates/conf/deployment.yaml.erb [hunk 11]
```diff
@@ -0,0 +1,581 @@
...
+          jdbcUrl: <%= p("wso2ei.sp_cluster_db.jdbc_url") %>
+          username: <%= p("wso2ei.sp_cluster_db.username") %>
+          password: <%= p("wso2ei.sp_cluster_db.password") %>
+          driverClassName: <%= p("wso2ei.sp_cluster_db.driver") %>
+          maxPoolSize: 50
+          idleTimeout: 60000
+          connectionTestQuery: <%= p("wso2ei.sp_cluster_db.query") %>
+          validationTimeout: 30000
+          isAutoCommit: false
+
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2ei_analytics_worker/templates/conf/deployment.yaml.erb:445]: Shall we remove the extra line here as well?

**Review by VimukthiPerera — APPROVED** [review]

**Pull request merged**